content/fast/reduce-network-payloads-using-text-compression/codelab-text-compression-brotli.md

```diff
@@ -0,0 +1,243 @@
+    })</strong>
+  ]
+}
+},
+</pre>
+
+The following arguments are used in the plugin array:
++  `asset`: The target asset name. Defaults to `'[path].br[query]'`
+ +  `[file]` is replaced with the original asset file name
+ +  `[query]` is replaced with the query
++  `test`: All assets that match this RegExp (i.e. javascript assets ending in `.js`) are processed
+
+When the app reloads and rebuilds, a compressed version of the main bundle is now created. Open the Glitch Console to take a look at what's inside the final `public/` directory that's served by the Node server.
+
+<div class="aside note">
+The <code>public/</code> directory is included in the <code>.gitignore</code> file. Directories that contain build files are usually included here in order to be ignored by Git, and Glitch also hides these files from the editor tree.
+</div>
+
++ Click the **Status** button.
+
+<web-screenshot type="status"></web-screenshot>
+
++ Click the **Console** button.
+
+<web-screenshot type="console"></web-screenshot>
+
++ In the console, run the following commands to change into the `public` directory
+and see all of its files:
+
+<pre class="devsite-terminal devsite-click-to-copy">
+cd public
+ls -lh
+</pre>
+
+<img class="screenshot" src="./console-static-compression-brotli.png"  alt="Bundle size with static Brotli compression">
+
+The brotli compressed version of the bundle, `main.bundle.js.br`, is now saved here as well and is **~76% smaller in size** (225 KB vs. 53 KB) than `main.bundle.js`.
+
+Next, tell the server to send these brotli-compressed files whenever their original JS versions are being requested. This can be done by defining a new route in `server.js` before the files are served with `express.static`.
+
+<pre class="prettyprint">
+var express = require('express');
+
+var app = express();
+
+<strong>app.get('*.js', (req, res, next) => {
+  req.url = req.url + '.br';
+  res.set('Content-Encoding', 'br');
+  next();
+});</strong>
+</pre>
+
+`app.get` is used to tell the server how to respond to a `GET` request for a specific endpoint. A callback function is then used to define how to handle this request. The route works like this:
++ Specifying `'*.js'` as the first argument means that this works for every endpoint that is fired to fetch a JS file.
++ Within the callback, `.br` is attached to the URL of the request and the `Content-Encoding` response header is set to `br`.
++ Finally, `next()` ensures that the sequence continues to any callback that may be next.
+
+Once the app reloads, take a look at the Network panel once more.
+
+<img class="screenshot" src="./network-static-compression-brotli.png"  alt="Network panel bundle size after static Brotli compression">
+
+Success! You have used Brotli compression to further compress your assets!
+
+## Conclusion
+
+This codelab illustrated how `brotli` can further reduce your app’s overall size. Where supported, `brotli` is a more powerful compression algorithm than `gzip`.
```

Comment:
Might be worth warning folks again that they should check if their CDN does this for them already.